api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -791,4 +792,43 @@ public void getCountOfDrugs_shouldReturnTheTotalNumberOfMatchingNumbers() {
 		assertEquals(new Integer(1),
 		    conceptService.getCountOfDrugs(phrase, conceptService.getConcept(conceptId), true, true, true));
 	}
+
+	/**
+	 * @see ConceptServiceImpl#saveConceptProposal(ConceptProposal) 
+	 */
+	@Test
+	public void saveConceptProposal_shouldReturnTheSavedConceptProposalObject() {
+		ConceptProposal conceptProposal = new ConceptProposal();
+		conceptProposal.setOriginalText("OriginalText");
+		List<ConceptProposal> existingConceptProposals = conceptService.getConceptProposals("OriginalText");
+		assertTrue(existingConceptProposals == null || existingConceptProposals.isEmpty());
+		ConceptProposal savedConceptProposal = conceptService.saveConceptProposal(conceptProposal);
+		assertEquals(conceptProposal, savedConceptProposal);
+	}
+
+	/**
+	 * @see ConceptServiceImpl#saveConceptProposal(ConceptProposal)
+	 */
+	@Test
+	public void saveConceptProposal_shouldReturnTheUpdatedConceptProposalObject() {
+		ConceptProposal conceptProposal = new ConceptProposal();
+		conceptProposal.setOriginalText("OriginalText");
+		ConceptProposal savedConceptProposal = conceptService.saveConceptProposal(conceptProposal);
+		savedConceptProposal.setOriginalText("AnotherOriginalText");
+		ConceptProposal updatedConceptProposal = conceptService.saveConceptProposal(savedConceptProposal);
+		assertEquals("AnotherOriginalText", updatedConceptProposal.getOriginalText());
+	}
+
+	/**
+	 * @see ConceptServiceImpl#saveConceptProposal(ConceptProposal)
+	 */
+	@Test
+	public void saveConceptProposal_shouldThrowIllegalArgumentExceptionWhenSavingNullConceptProposal() {
+		try {
+			conceptService.saveConceptProposal(null);
+			fail("IllegalArgumentException not thrown");
```

Comment:
This is not the way to assert an Exception in java. If you need to check whether its throwing an perticular exception , you can use "expected" annotation for that

Reply:
Thanks for the tips, PR amended.